### DISCUSSION TITLE
如何安装在openwrt上

### DISCUSSION THREAD

**Discussion by luoling76** [Netatalk/netatalk#562] [Q&A] [answered]
如何安装在openwrt上？

**Comment by rdmark** [comment]
@luoling76 If I understand the question correctly, you want to run Netatalk on OpenWrt?

If the Wikipedia article is correct, OpenWrt is Linux based and comes with gcc and autotools toolchain. You should be able to build it from source if you can satisfy all of the mandatory software requirements, notably BerkeleyDB.

Please see the manual for more info! https://netatalk.sourceforge.io/3.1/htmldocs/installation.html

**Reply by luoling76** [reply]
对，你理解的对了，但是我下载下来的压缩包，解压了是一堆文件和文件夹。不会构建openwrt所需要的安装包ipk。如果有一键安装的脚本更好了！

**Reply by rdmark** [reply]
I agree that having a pre-built package for your OS is ideal. In the Netatalk project we rely on external volunteers to create and maintain such packages, e.g. deb for Debian and rpm for Red Hat.

Perhaps you can ask in the OpenWrt community or adjacent Embedded Linux projects if someone there would be able to package Netatalk in the ipk format?

**Comment by rdmark** [comment, marked as answer]
There are now select OpenWrt netatalk ipks available. See https://github.com/Netatalk/netatalk/discussions/1694
